File: array/anagram.cpp
#include<bits/stdc++.h>
using namespace std;
#define endl "\n"


bool isAnagram(string s, string t) {
    unordered_map<char,int>dict;
    if(s.length()!=t.length())
        return false;
    int i=0,j=0;

    while(i<s.length()){
        dict[s[i]]++;
        dict[t[i]]--;
        i++;
    }
    for(auto x:dict){
        if(x.second!=0){
            return false;
        }
    };
    return true;
        
}


int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    string s ="anagram" ,t="nagaram";
    cout<<isAnagram(s,t);
    return 0;
}

File: js/anagram.js
let first = "rat";
let second = "tar";


var isAnagram = function(first, second) {
    let mp = {};
    if (first.length != second.length) return false;
    for (let i = 0; i < first.length; i++) {
        if (mp[first[i]]) {
            mp[first[i]]++;
        } else {
            mp[first[i]] = 1;
        }
    }
    for (let i = 0; i < second.length; i++) {
        if (mp[second[i]]) {
            mp[second[i]]--;
        } else {
            return false;
        }
    }
    for (let i in mp) {
        if (mp[i] != 0) return false;
    }
    return true;
};


console.log(isAnagram(first, second ));